docs/widgets/data_table.md

```diff
@@ -20,11 +24,143 @@ The example below populates a table with CSV data.
     --8<-- "docs/examples/widgets/data_table.py"
     ```
 
+To add rows and columns one-at-a-time, use [add_row][textual.widgets.DataTable.add_row] and [add_column][textual.widgets.DataTable.add_column].
+
+#### Styling and justifying cells
+
+Cells can contain more than just plain strings - [Rich](https://rich.readthedocs.io/en/stable/introduction.html) renderables such as [`Text`](https://rich.readthedocs.io/en/stable/text.html?highlight=Text#rich-text) are also supported.
+`Text` objects provide an easy way to style and justify cell content:
+
+=== "Output"
+
+    ```{.textual path="docs/examples/widgets/data_table_renderables.py"}
+    ```
+
+=== "data_table_renderables.py"
+
+    ```python
+    --8<-- "docs/examples/widgets/data_table_renderables.py"
+    ```
+
+### Keys
+
+When adding a row to the table, you can supply a _key_ to [add_row][textual.widgets.DataTable.add_row].
+A key is a unique identifier for that row.
+If you don't supply a key, Textual will generate one for you and return it from `add_row`.
+This key can later be used to reference the row, regardless of its current position in the table.
+
+When working with data from a database for example, you make wish to set the row `key` to the primary key of the data to ensure uniqueness.
+The [add_column][textual.widgets.DataTable.add_column] also accepts a `key` argument and works similarly.
+
+Keys are important because cells in a data table can change location due factors like row deletion and sorting.
+Thus, using keys instead of coordinates allows us to refer to data without worrying about its current location in the table.
+
+Sometimes you wish to perform an operation on a cell/row/column based solely on coordinates, without regard for the data that is currently visible at those coordinates.
+In these cases you can make use of the [coordinate_to_cell_key][textual.widgets.DataTable.coordinate_to_cell_key] method to convert
+a coordinate to a _cell key_, which is a `(row_key, column_key)` pair.
+
+### Cursors
+
+You can retrieve the current coordinate of the cursor from the `cursor_coordinate` reactive attribute.
```

Comment:
```suggestion
The coordinate of the cursor is exposed via the `cursor_coordinate` reactive attribute.
```

Original wording suggests to me that there may be some additional step involved.